### ISSUE TITLE
Copy from container will create directory

### ISSUE THREAD

**Issue by dmanners** [markshust/docker-magento#191]
**Description**

It seems that the `bin/copyfromcontainer` script will always, unless --all is used, try to create a folder with the name passed in as argument. For most cases I think this is fine but for example this does not work for `bin/dev-urn-catalog-generate`.

On my host I end up with a `misc.xml` folder and the error message `cannot overwrite directory "/home/dmanners/project/magento2-github/src/misc.xml" with non-directory "/home/dmanners/project/magento2-github/src"`

**Steps To Reproduce**
1. Have a running system
1. Run `bin/dev-urn-catalog-generate` or `bin/copyfromcontainer singlefile`

**Expected Result**
1. File copied over and no error message displayed

**Actual Result**
1. Directory with the same name as the file is created on the host
2. Error message shown: `cannot overwrite directory "/home/dmanners/project/magento2-github/src/misc.xml" with non-directory "/home/dmanners/project/magento2-github/src"`

**Comment by Seb2nim** [comment]
I confirm same issue.
Apparently recent update made `bin/copyfromcontainer` / `bin/copytocontainer` applicatble only to directories...

I also suggest these scripts to be able to handle "src" being a symlink to actual project.

`bin/copyfromcontainer`
```
#!/bin/bash
[ -z "$1" ] && echo "Please specify a directory or file to copy from container (ex. vendor, --all)" && exit

REAL_SRC=$(cd -P "src" && pwd)
if [ "$1" == "--all" ]; then
  docker cp $(docker-compose ps -q phpfpm|awk '{print $1}'):/var/www/html/./ $REAL_SRC/
  echo "Completed copying all files from container to host"
else
  docker cp $(docker-compose ps -q phpfpm|awk '{print $1}'):/var/www/html/$1 $REAL_SRC/
  echo "Completed copying $1 from container to host"
fi
```

`bin/copytocontainer`
```
#!/bin/bash
[ -z "$1" ] && echo "Please specify a directory or file to copy to container (ex. vendor, --all)" && exit

REAL_SRC=$(cd -P "src" && pwd)
if [ "$1" == "--all" ]; then
  docker cp $REAL_SRC/./ $(docker-compose ps -q phpfpm|awk '{print $1}'):/var/www/html/
  echo "Completed copying all files from host to container"
  bin/fixowns
  bin/fixperms
else
  docker cp $REAL_SRC/$1 $(docker-compose ps -q phpfpm|awk '{print $1}'):/var/www/html/$1
  echo "Completed copying $1 from host to container"
  bin/fixowns $1
  bin/fixperms $1
fi
```

**Comment by markshust** [comment]
Hi @Seb2nim -- merged in https://github.com/markshust/docker-magento/pull/198, does that resolve this issue as well?

Also, I apologize but I'm a bit confused about what each ticket is and if they are all fixed by this PR?

https://github.com/markshust/docker-magento/issues/191
https://github.com/markshust/docker-magento/issues/194
https://github.com/markshust/docker-magento/issues/195

Are these all the same issue?

**Comment by Seb2nim** [comment]
Hi @markshust, 
I agree, pretty confusing... summary ;
#191 and #194 solved by PR #198 

#195 Is related to urn generate and path not converted (sed replacement). Fixed by PR #197.

**Comment by markshust** [comment]
Thanks! Both of those PR have been merged, and I just tagged version 26.0.0. This should now be resolved.

https://github.com/markshust/docker-magento/releases/tag/26.0.0
